### PR TITLE
Handled nested structures for TensorFlow backend for keras.ops.map. 

### PR DESCRIPTION
**Pull request by mehtamansi29**
Handling nested structure for keras.ops.map in tensorflow backend. Fixes: [#20048](https://github.com/keras-team/keras/issues/20048)

### CODE REVIEW

File: keras/src/backend/tensorflow/core.py [hunk 1]
```diff
@@ -237,9 +237,14 @@ def map(f, xs):
     def get_fn_output_signature(x):
         out = f(x)
         return tree.map_structure(tf.TensorSpec.from_tensor, out)
-
-    fn_output_signature = get_fn_output_signature(xs[0])
-    return tf.map_fn(f, xs, fn_output_signature=fn_output_signature)
+    
+    if isinstance(xs,dict) or isinstance(xs,list) or isinstance(xs,tuple):
```
Comment (james77777778) [inline keras/src/backend/tensorflow/core.py:241]: I believe this line could be replaced with `tree.is_nested(xs)`.

Comment (mehtamansi29) [reply keras/src/backend/tensorflow/core.py:241]: @james77777778 - Let me check with same and update here if it will works fine.

**Review by fchollet — COMMENTED** [review]
Thanks for the PR!

You can format the code by running `sh shell/format.sh`.

Please add a unit test for this case.

**Comment by fchollet** [comment]
It looks like the unit test is failing...?

File: keras/src/backend/tensorflow/nestedstructure_test.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+            tf.convert_to_tensor(np.random.rand(4, 100, 3), dtype=tf.float32),
+            tf.convert_to_tensor(
+                np.random.randint(0, 10, size=(4, 1)), dtype=tf.int32
+            ),
+        ]
+
+    def test_dict_input_fn_outputs(self):
+        ys = keras.ops.map(dict_input_fn, self.xs)
+        self.assertEqual(ys["x"].shape, (4, 100))
+        self.assertTrue((ys["y"] == self.xs["y"] + 1).numpy().all())
```
Comment (fchollet) [inline keras/src/backend/tensorflow/nestedstructure_test.py:44]: You should use `ops.convert_to_numpy()` instead of `.numpy()`

**Review by fchollet — COMMENTED** [review]
Thanks for the update -- please fix the tests!

**Review by fchollet — APPROVED** [review]
Will take it from here. Thanks for the PR!

**Pull request merged**